Shared/Supporting Files/Extensions/CategoryView+Search.swift

```diff
@@ -14,14 +14,44 @@
 
 import SwiftUI
 
-extension CategoryView {
-    /// Searches the sample list using the query.
-    /// - Returns: A set of samples matching the query criteria.
-    func searchSamples() -> [Sample] {
-        if query.isEmpty {
-            return samples
-        } else {
-            return samples.filter { $0.name.localizedCaseInsensitiveContains(query) }
+extension CategoryView {    
+    /// Searches the samples using the sample's name and the query.
+    /// - Returns: The samples whose name partially matches the query.
+    func searchSamplesNames() -> [Sample] {
+        // Return all the samples when query is empty.
+        guard !query.isEmpty else { return samples }
+        
+        // Perform a partial text search using the sample's name and the query.
+        let nameSearchResults = samples.filter { sample in
+            sample.name.localizedCaseInsensitiveContains(query)
         }
+        previousSearchResults.formUnion(nameSearchResults.map { $0.name })
```

Comment:
```suggestion
        previousSearchResults.formUnion(nameSearchResults.map(\.name))
```
Can use keypath here.